pkg/rpcservice/listen.go

```diff
@@ -0,0 +1,200 @@
+package rpcservice
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/stripe/stripe-cli/pkg/proxy"
+	"github.com/stripe/stripe-cli/pkg/websocket"
+	"github.com/stripe/stripe-cli/rpc"
+)
+
+const webhooksWebSocketFeature = "webhooks"
+
+// IProxy enables mocking a proxy object in tests
+type IProxy interface {
+	Run(context.Context) error
+}
+
+var createProxy = func(cfg *proxy.Config) (IProxy, error) {
+	return proxy.Init(cfg)
+}
+
+// Listen returns a stream of webhook events and forwards them to a local endpoint
+func (srv *RPCService) Listen(req *rpc.ListenRequest, stream rpc.StripeCLI_ListenServer) error {
+	deviceName, err := srv.cfg.UserCfg.Profile.GetDeviceName()
+	if err != nil {
+		return err
+	}
+
+	key, err := srv.cfg.UserCfg.Profile.GetAPIKey(req.Live)
+	if err != nil {
+		return err
+	}
+
+	// validate forward-urls args
```

Comment:
This seems like it is proxy-specific validation and should be handling by the proxy package (maybe directly by proxy.Init) instead of gRPC?

Reply:
Agreed, it should be in proxy.Init!